### PR TITLE
Add "INSTALL_SOURCES" flag to base Dockerfile

### PR DESCRIPTION
**Pull request by itrushkin**
Allows disabling sources installation for lightweight docker images.

### CODE REVIEW

**Review by alexey-gruzdev — APPROVED** [review]

**Review by alexey-khorkin — APPROVED** [review]

File: openfl-docker/Dockerfile.base [hunk 1]
```diff
@@ -42,7 +44,14 @@ COPY . .
 # Install pip
 RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1
 RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && python get-pip.py && rm -rf get-pip.py
-RUN pip install --no-cache-dir  .
+RUN pip install --no-cache-dir . pip-licenses
```
Comment (aleksandr-mokrov) [inline openfl-docker/Dockerfile.base]: Probably we should install pip-licenses only `"$INSTALL_SOURCES" = "yes"` and move `RUN pip install --no-cache-dir pip-licenses` and `WORKDIR /thirdparty` under statement `RUN if [ "$INSTALL_SOURCES" = "yes" ]; then`?

**Review by aleksandr-mokrov — APPROVED** [review]

File: openfl-docker/Dockerfile.base (hunk 1 above)
Comment (itrushkin) [reply openfl-docker/Dockerfile.base]: Yes, installation of pip-licenses can be moved under if-statement. But regarding `WORKDIR`, if-statement cannot be separated from the `RUN` command, so we only can change directory before and after the `RUN` command.

**Comment by alexey-gruzdev** [comment]
Jenkins, please retry a build

**Comment by alexey-gruzdev** [comment]
Jenkins please retry a build

**Comment by alexey-gruzdev** [comment]
Jenkins please retry a build

**Pull request merged**